File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let max = 0;
  let minPrice = Number.MAX_SAFE_INTEGER;
  for (let i = 0; i < prices.length; i++) {
      if (prices[i] < minPrice) {
          minPrice = prices[i];
      } else if (prices[i] - minPrice > max) {
          max = prices[i] - minPrice;
      }
  }
  return max;
};

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
import "math"

func maxProfit(prices []int) int {
	max := 0
	minPrice := math.MaxInt32
	for _, price := range prices {
		if price < minPrice {
			minPrice = price
		} else if price-minPrice > max {
			max = price - minPrice
		}
	}
	return max
}